pkg/minikube/cluster/cluster.go

```diff
@@ -315,3 +318,58 @@ func CreateSSHShell(api libmachine.API, args []string) error {
 	}
 	return client.Shell(strings.Join(args, " "))
 }
+
+func GetDashboardURL(api libmachine.API) (string, error) {
+	host, err := checkIfApiExistsAndLoad(api)
+	if err != nil {
+		return "", err
+	}
+
+	ip, err := host.Driver.GetIP()
+	if err != nil {
+		return "", err
+	}
+
+	port, err := getDashboardPort()
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("http://%s:%d", ip, port), nil
+}
+
+type serviceGetter interface {
+	Get(name string) (*api.Service, error)
+}
+
+func getDashboardPort() (int, error) {
+	services, err := getKubernetesServicesWithNamespace("kube-system")
+	if err != nil {
+		return 0, err
+	}
+	return getDashboardPortFromServiceGetter(services)
+}
+
+func getDashboardPortFromServiceGetter(services serviceGetter) (int, error) {
+	dashboardService, err := services.Get("kubernetes-dashboard")
+	if err != nil {
+		return 0, fmt.Errorf("Error getting kubernetes-dashboard service: %s", err)
+	}
+	return int(dashboardService.Spec.Ports[0].NodePort), nil
+}
+
+func getKubernetesServicesWithNamespace(namespace string) (serviceGetter, error) {
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
```

Comment:
Nit: you could inline these two into the NewNonInteractiveDeferredLoadingClientConfig...